File: list_businesses.swift
#!/usr/bin/env swift

import Foundation
import FirebaseCore
import FirebaseFirestore
import FirebaseAuth

// Simple script to list all businesses in Firebase
// This uses the existing FirebaseClient logic

struct Business {
    let id: String
    let name: String
    let businessType: String
    let address: String?
    let phone: String?
    let website: String?
    let ownerId: String
    let isActive: Bool
    let verificationStatus: String
    let createdAt: Date
}

class BusinessLister {
    private let db = Firestore.firestore()
    
    func listAllBusinesses() async throws -> [Business] {
        print("🔍 Querying Firebase for all businesses...")
        
        // Query both restaurants and businesses collections
        let restaurantsSnap = try await db.collection("restaurants")
            .order(by: "name", descending: false)
            .getDocuments()
        
        let businessesSnap = try await db.collection("businesses")
            .order(by: "name", descending: false)
            .getDocuments()
        
        var allBusinesses: [Business] = []
        
        print("📊 Found \(restaurantsSnap.documents.count) restaurants and \(businessesSnap.documents.count) businesses")
        
        // Process restaurants collection
        for doc in restaurantsSnap.documents {
            let data = doc.data()
            guard let name = data["name"] as? String else { continue }
            
            let createdAtTimestamp = data["createdAt"] as? Timestamp ?? Timestamp()
            
            let business = Business(
                id: doc.documentID,
                name: name,
                businessType: "restaurant",
                address: data["address"] as? String,
                phone: data["phone"] as? String,
                website: data["website"] as? String,
                ownerId: data["ownerId"] as? String ?? "",
                isActive: data["isActive"] as? Bool ?? true,
                verificationStatus: data["verificationStatus"] as? String ?? "pending",
                createdAt: createdAtTimestamp.dateValue()
            )
            allBusinesses.append(business)
        }
        
        // Process businesses collection
        for doc in businessesSnap.documents {
            let data = doc.data()
            guard let name = data["name"] as? String else { continue }
            
            let createdAtTimestamp = data["createdAt"] as? Timestamp ?? Timestamp()
            let businessType = data["businessType"] as? String ?? "other"
            
            let business = Business(
                id: doc.documentID,
                name: name,
                businessType: businessType,
                address: data["address"] as? String,
                phone: data["phone"] as? String,
                website: data["website"] as? String,
                ownerId: data["ownerId"] as? String ?? "",
                isActive: data["isActive"] as? Bool ?? true,
                verificationStatus: data["verificationStatus"] as? String ?? "pending",
                createdAt: createdAtTimestamp.dateValue()
            )
            allBusinesses.append(business)
        }
        
        return allBusinesses.sorted { $0.name < $1.name }
    }
    
    func printBusinessList(_ businesses: [Business]) {
        print("\n" + "="*80)
        print("📋 ALL BUSINESSES IN FIREBASE DATABASE")
        print("="*80)
        print("Total: \(businesses.count) businesses\n")
        
        for (index, business) in businesses.enumerated() {
            print("[\(index + 1)] \(business.name)")
            print("    Type: \(business.businessType.capitalized)")
            print("    ID: \(business.id)")
            if let address = business.address {
                print("    Address: \(address)")
            }
            if let phone = business.phone {
                print("    Phone: \(phone)")
            }
            if let website = business.website {
                print("    Website: \(website)")
            }
            print("    Owner ID: \(business.ownerId)")
            print("    Status: \(business.isActive ? "Active" : "Inactive")")
            print("    Verification: \(business.verificationStatus.capitalized)")
            
            let formatter = DateFormatter()
            formatter.dateStyle = .medium
            formatter.timeStyle = .short
            print("    Created: \(formatter.string(from: business.createdAt))")
            print("")
        }
        
        print("="*80)
        print("Summary by Type:")
        let groupedByType = Dictionary(grouping: businesses) { $0.businessType }
        for (type, businesses) in groupedByType.sorted(by: { $0.key < $1.key }) {
            print("  \(type.capitalized): \(businesses.count)")
        }
        
        print("\nSummary by Status:")
        let activeCount = businesses.filter { $0.isActive }.count
        let inactiveCount = businesses.count - activeCount
        print("  Active: \(activeCount)")
        print("  Inactive: \(inactiveCount)")
        
        print("\nSummary by Verification:")
        let verificationGroups = Dictionary(grouping: businesses) { $0.verificationStatus }
        for (status, businesses) in verificationGroups.sorted(by: { $0.key < $1.key }) {
            print("  \(status.capitalized): \(businesses.count)")
        }
        print("="*80)
    }
}

// Main execution
Task {
    do {
        let lister = BusinessLister()
        let businesses = try await lister.listAllBusinesses()
        lister.printBusinessList(businesses)
    } catch {
        print("❌ Error listing businesses: \(error)")
        print("Error details: \(error.localizedDescription)")
    }
}

// Keep the script running
RunLoop.main.run()

File: list_businesses.js
const admin = require('firebase-admin');
const fs = require('fs');

// Initialize Firebase Admin SDK
const serviceAccount = require('./functions/service-account-key.json');

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  databaseURL: 'https://kevin-ios-app-default-rtdb.firebaseio.com'
});

const db = admin.firestore();

async function listAllBusinesses() {
  try {
    console.log('🔍 Querying Firebase for all businesses...\n');
    
    // Query restaurants collection
    const restaurantsSnapshot = await db.collection('restaurants').orderBy('name').get();
    
    // Query businesses collection (if it exists)
    let businessesSnapshot;
    try {
      businessesSnapshot = await db.collection('businesses').orderBy('name').get();
    } catch (error) {
      console.log('ℹ️  No businesses collection found, only checking restaurants');
      businessesSnapshot = { docs: [] };
    }
    
    const allBusinesses = [];
    
    // Process restaurants
    restaurantsSnapshot.forEach(doc => {
      const data = doc.data();
      allBusinesses.push({
        id: doc.id,
        name: data.name || 'Unknown',
        type: 'restaurant',
        address: data.address || 'No address',
        phone: data.phone || 'No phone',
        website: data.website || 'No website',
        ownerId: data.ownerId || 'No owner',
        isActive: data.isActive !== false,
        verificationStatus: data.verificationStatus || 'pending',
        placeId: data.placeId || 'No Google Place ID',
        latitude: data.latitude || 'No coordinates',
        longitude: data.longitude || 'No coordinates',
        createdAt: data.createdAt ? data.createdAt.toDate() : new Date(),
        collection: 'restaurants'
      });
    });
    
    // Process businesses
    businessesSnapshot.docs.forEach(doc => {
      const data = doc.data();
      allBusinesses.push({
        id: doc.id,
        name: data.name || 'Unknown',
        type: data.businessType || 'other',
        address: data.address || 'No address',
        phone: data.phone || 'No phone',
        website: data.website || 'No website',
        ownerId: data.ownerId || 'No owner',
        isActive: data.isActive !== false,
        verificationStatus: data.verificationStatus || 'pending',
        placeId: data.placeId || 'No Google Place ID',
        latitude: data.latitude || 'No coordinates',
        longitude: data.longitude || 'No coordinates',
        createdAt: data.createdAt ? data.createdAt.toDate() : new Date(),
        collection: 'businesses'
      });
    });
    
    // Sort by name
    allBusinesses.sort((a, b) => a.name.localeCompare(b.name));
    
    console.log('=' .repeat(80));
    console.log('📋 ALL BUSINESSES IN FIREBASE DATABASE');
    console.log('=' .repeat(80));
    console.log(`Total: ${allBusinesses.length} businesses\n`);
    
    allBusinesses.forEach((business, index) => {
      console.log(`[${index + 1}] ${business.name}`);
      console.log(`    Type: ${business.type.charAt(0).toUpperCase() + business.type.slice(1)}`);
      console.log(`    Collection: ${business.collection}`);
      console.log(`    ID: ${business.id}`);
      console.log(`    Address: ${business.address}`);
      console.log(`    Phone: ${business.phone}`);
      console.log(`    Website: ${business.website}`);
      console.log(`    Owner ID: ${business.ownerId}`);
      console.log(`    Status: ${business.isActive ? 'Active' : 'Inactive'}`);
      console.log(`    Verification: ${business.verificationStatus.charAt(0).toUpperCase() + business.verificationStatus.slice(1)}`);
      console.log(`    Google Place ID: ${business.placeId}`);
      console.log(`    Coordinates: ${business.latitude}, ${business.longitude}`);
      console.log(`    Created: ${business.createdAt.toLocaleDateString()} ${business.createdAt.toLocaleTimeString()}`);
      console.log('');
    });
    
    console.log('=' .repeat(80));
    console.log('📊 SUMMARY BY TYPE:');
    const typeGroups = allBusinesses.reduce((acc, business) => {
      acc[business.type] = (acc[business.type] || 0) + 1;
      return acc;
    }, {});
    
    Object.entries(typeGroups)
      .sort(([a], [b]) => a.localeCompare(b))
      .forEach(([type, count]) => {
        console.log(`  ${type.charAt(0).toUpperCase() + type.slice(1)}: ${count}`);
      });
    
    console.log('\n📊 SUMMARY BY STATUS:');
    const activeCount = allBusinesses.filter(b => b.isActive).length;
    const inactiveCount = allBusinesses.length - activeCount;
    console.log(`  Active: ${activeCount}`);
    console.log(`  Inactive: ${inactiveCount}`);
    
    console.log('\n📊 SUMMARY BY VERIFICATION:');
    const verificationGroups = allBusinesses.reduce((acc, business) => {
      acc[business.verificationStatus] = (acc[business.verificationStatus] || 0) + 1;
      return acc;
    }, {});
    
    Object.entries(verificationGroups)
      .sort(([a], [b]) => a.localeCompare(b))
      .forEach(([status, count]) => {
        console.log(`  ${status.charAt(0).toUpperCase() + status.slice(1)}: ${count}`);
      });
    
    console.log('\n📊 SUMMARY BY COLLECTION:');
    const collectionGroups = allBusinesses.reduce((acc, business) => {
      acc[business.collection] = (acc[business.collection] || 0) + 1;
      return acc;
    }, {});
    
    Object.entries(collectionGroups).forEach(([collection, count]) => {
      console.log(`  ${collection}: ${count}`);
    });
    
    console.log('=' .repeat(80));
    
  } catch (error) {
    console.error('❌ Error listing businesses:', error);
    console.error('Error details:', error.message);
  }
}

// Run the query
listAllBusinesses().then(() => {
  console.log('✅ Query completed');
  process.exit(0);
}).catch(error => {
  console.error('❌ Script failed:', error);
  process.exit(1);
});
